frontend/src/components/VersionSelector/VersionSelector.tsx

```diff
@@ -34,47 +33,67 @@ interface VersionSelectorProps {
  * ```
  *  */
 export const VersionSelector: FC<VersionSelectorProps> = ({
+  currentVersionUUID,
   versions,
   onVersionChange,
 }) => {
   const id = useId();
   const headingId = `version-selector-heading-${id}`;
+  const [isExpanded, setIsExpanded] = useState(false);
 
   const today = new Date();
 
-  const sortedVersions = [...versions].sort(
-    (a, b) =>
-      new Date(a.beginGeldigheid).getTime() -
-      new Date(b.beginGeldigheid).getTime(),
+  // Versions sorted by date.
+  const sortedVersions = useMemo(
+    () =>
+      [...versions].sort(
+        (a, b) =>
+          new Date(a.beginGeldigheid).getTime() -
+          new Date(b.beginGeldigheid).getTime(),
+      ),
+    [versions],
   );
 
-  const currentVersion = sortedVersions.find((v) => {
-    const beginDate = new Date(v.beginGeldigheid);
-    const endDate = v.eindeGeldigheid ? new Date(v.eindeGeldigheid) : null;
-
-    return !v.concept && beginDate <= today && (!endDate || endDate > today);
-  });
-
-  const conceptVersion = sortedVersions.filter((v) => v.concept)[0];
+  // The current active (but not necessarily selected) versions
+  const currentVersion = useMemo(
+    () =>
+      sortedVersions.find((v) => {
+        const beginDate = new Date(v.beginGeldigheid);
+        const endDate = v.eindeGeldigheid ? new Date(v.eindeGeldigheid) : null;
+
+        return (
+          !v.concept && beginDate <= today && (!endDate || endDate > today)
+        );
+      }),
+    [sortedVersions],
+  );
 
-  const [isExpanded, setIsExpanded] = useState(false);
-  const [selectedVersionId, setSelectedVersionId] = useState<
-    string | undefined
-  >(currentVersion?.uuid || conceptVersion?.uuid);
+  // The (last) concept version, we assume there should be max 1.
+  const conceptVersion = useMemo(() => {
+    const concepts = sortedVersions.filter((v) => v.concept);
+    return concepts[concepts.length - 1];
+  }, [sortedVersions]);
 
-  const handleVersionChange = (version: ZaakTypeVersion) => {
-    setSelectedVersionId(version.uuid);
+  const handleVersionChange = (
+    version: components["schemas"]["VersionSummary"],
+  ) => {
     onVersionChange?.(version);
   };
 
-  const getVisibleVersions = () => {
-    const isHistorical = (v: ZaakTypeVersion) =>
-      !v.concept &&
-      new Date(v.beginGeldigheid) < today &&
-      (v.eindeGeldigheid ? new Date(v.eindeGeldigheid) <= today : false);
-
-    const historicalVersions = sortedVersions.filter(isHistorical);
+  // Historical versions.
+  const historicalVersions = useMemo(
+    () =>
+      sortedVersions.filter(
+        (v) =>
+          !v.concept &&
+          new Date(v.beginGeldigheid) < today &&
+          (v.eindeGeldigheid ? new Date(v.eindeGeldigheid) <= today : false),
+      ),
```

Comment:
Aren't the historical versions all the sorted non-concept versions except the last one (depending on whether its eindeGeldigheid is < today)?